presto-docs/src/main/sphinx/functions/ml.rst

```diff
@@ -0,0 +1,151 @@
+from the existing numerical values, :func:`features`::
+
+    SELECT features(1.0, 2.0, 3.0) features;
+
+.. code-block:: none
+
+           features
+    -----------------------
+     {0=1.0, 1=2.0, 2=3.0}
+
+The output from :func:`features` can be directly passed to ML functions.
+
+Classification
+---------------
+
+Classification is a type of supervised learning problem to predict the distinct
+label from the given feature vector. The interface looks close to what in
+Teradata Aster or `BigQuery ML <https://cloud.google.com/bigquery-ml/docs/bigqueryml-intro>`_. :func:`learn_classifier` constructs the SVM
+model from the sequence of pairs of labels and features::
+
+    SELECT
+      learn_classifier(species, features(sepal_length, sepal_width, petal_length, petal_width))
+    FROM
+      iris
+
+It returns the trained model in a serialized format.
+
+.. code-block:: none
+
+                          model
+    -------------------------------------------------
+     3c 43 6c 61 73 73 69 66 69 65 72 28 76 61 72 63
+     68 61 72 29 3e
+
+:func:`classify` returns the predicted label by using the trained model. For now, we
+cannot save the model natively. It needs to be passed in the format of nested query.::
+
+    SELECT
+      classify(features(5.9, 3, 5.1, 1.8), model) predicted_label
+    FROM (
+      SELECT
+        learn_classifier(species, features(sepal_length, sepal_width, petal_length, petal_width)) model
+      FROM
+        iris
+    ) t
+
+.. code-block:: none
+
+     predicted_label
+    -----------------
+     Iris-virginica
+
+That means we need to run the training process at the same time when predicting values.
+Internally, the model is trained by `libsvm <https://www.csie.ntu.edu.tw/~cjlin/libsvm/>`_. If you want to
+control the internal parameters of the model thoroughly, :func:`learn_libsvm_classifier` would be helpful.
+
+Regression
+-----------
+
+Regression is another type of supervised learning problem predicting continuous
+value, unlike the classification problem. The target must be a numerical value that can
+be described as :ref:`double`::
+
+The following code shows the creation of the model predicting ``sepal_length``
+from the other 3 features.::
+
+    SELECT
+      learn_regressor(sepal_length, features(sepal_width, petal_length, petal_width)) as model
+    FROM
+      iris
+
+The way to use the model is similar to the classification case.::
+
+    SELECT
+      regress(features(3, 5.1, 1.8), model) predicted_target
+    FROM (
+      SELECT
+        learn_regressor(sepal_length, features(sepal_width, petal_length, petal_width)) as model
+      FROM iris
+    ) t;
+
+.. code-block:: none
+
+     predicted_target
+    -------------------
+     6.407376822560477
+
+Internally, the model is also trained by `libsvm <https://www.csie.ntu.edu.tw/~cjlin/libsvm/>`_.
+:func:`learn_libsvm_regressor` provides you a way to control the training process flexibly.
+
+Machine Learning Functions
+--------------------------
```

Comment:
All these functions should be added to the index as well. https://github.com/prestosql/presto/blob/master/presto-docs/src/main/sphinx/functions/list.rst